app/models/publication.rb

```diff
@@ -43,31 +43,7 @@ class Publication < ActiveRecord::Base
     autosave: true,
     dependent: :destroy,
     after_add: :pubhash_needs_update!,
-    after_remove: :pubhash_needs_update! do
-    # See also: update_any_new_contribution_info_in_pub_hash_to_db
-    def build_or_update(author, contribution_hash = {})
-      # Assign or update these contribution attributes
-      # Ensure the contribution attributes contain the right author
-      contribution_hash[:author_id] = author.id
-      contribution_hash[:cap_profile_id] = author.cap_profile_id if author.cap_profile_id.present?
-      contrib = find_or_initialize_by(author_id: author.id) do |new_contrib|
-        new_contrib.assign_attributes(contribution_hash)
-        new_contrib.publication = proxy_association.owner
-      end
-      if contrib.persisted?
-        # Update an existing contribution
-        contrib.update(contribution_hash)
-      else
-        contrib.save
-        # SELF is an array of Contributions
-        self << contrib unless include? contrib
-      end
-      # The `proxy_association.owner` is a Publication instance, so
-      # set a trigger that will force it to update it's pub_hash data.
-      proxy_association.owner.pubhash_needs_update!
-      contrib
-    end
-  end
+    after_remove: :pubhash_needs_update!
```

Comment:
- Removing this seems to intersect with a prior commit that changed the contrib, maybe drop that prior commit?
- It's difficult to see that moving this to the api code has captured all the logic accurately, like what about checking if the cap_profile_id is present or not?
- The `after_remove` shouldn't be touched here - looks like accidental whitespace change.

Reply:
The `do` part of the `after_remove` line was removed (i.e. the start of the block now removed).

Reply:
The lines moved to here:
https://github.com/sul-dlss/sul_pub/pull/334/files#diff-c6ef41c2e0bc46fa80c88d0c5ccb026bR27

Correct me if I'm wrong, but we're assigning the contribution the author's ID (as `author_id`), so my sense is it would be counterproductive to simultaneously retain some *other* author's `cap_profile_id`.  This author is taking control of the contrib, so it should be tied to their respective IDs (or nobody's).

Reply:
IIRC (not sure), basically we don't trust the CAP-team to call our API with the same params all the time and, maybe, sometimes they have our `author.id` (as something like `sul_author_id`) and sometimes they don't but they send the `cap_profile_id`; and sometimes they only send one or the other; also if we only get a `cap_profile_id` and we can't find an author.... we try to place nicely.  I dunno, maybe I'm being paranoid or there have been weird bugs fixed at some point.  Maybe I just need to get dirty with this in my console to see what's going on.

Reply:
Those values aren't coming straight from query params though, they are coming out of a valid `Author` object in the database.  So we can trust them.